cookbook/upgrade/bundles.rst

```diff
@@ -0,0 +1,155 @@
+.. index::
+    single: Upgrading; Bundle; Major Version
+
+Upgrading a Third-Party Bundle for a Major Symfony Version
+==========================================================
+
+Symfony 3 was released on November 2015. Although this version doesn't contain
+any new feature, it removes all the backwards compatibility layers included in
+the previous 2.8 version. If your bundle uses any deprecated feature and it's
+published as a third-party bundle, applications upgrading to Symfony 3 will no
+longer be able to use it.
+
+Allow to Install Symfony 3 Components
+-------------------------------------
+
+Most third-party bundles define their Symfony dependencies using the ``~2.N`` or
+``^2.N`` constraints in the ``composer.json`` file. For example:
+
+.. code-block:: json
+
+    {
+        "require": {
+            "symfony/framework-bundle": "~2.3",
+            "symfony/finder": "~2.3",
+            "symfony/validator": "~2.3"
+        }
+    }
+
+These constraints prevent the bundle from using Symfony 3 components, so it makes
+it impossible to install it in a Symfony 3 based application. This issue is very
+easy to solve thanks to the flexibility of Composer dependencies constraints.
+Just replace ``~2.N`` by ``~2.N|~3.0`` (or ``^2.N`` by ``^2.N|~3.0``).
+
+The above example can be updated to work with Symfony 3 as follows:
+
+.. code-block:: json
+
+    {
+        "require": {
+            "symfony/framework-bundle": "~2.3|~3.0",
+            "symfony/finder": "~2.3|~3.0",
+            "symfony/validator": "~2.3|~3.0"
+        }
+    }
+
+Look for Deprecations and Fix Them
+----------------------------------
+
+Besides allowing to install Symfony 3 component, your bundle must stop using
+any feature deprecated in 2.8 version, because they'll throw exceptions in 3.0
+version. The easiest way to detect deprecations is to install the `PHPUnit Bridge`_
```

Comment:
Not throw exceptions, they are just removed (in some cases resulting in an exception, in most cases resulting in a PHP error).